File: backend/app/schemas/transcript.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class TranscriptSegmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    speaker_name: str
    start_time_ms: int
    end_time_ms: int
    text: str
    sequence_index: int


class TranscriptResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    meeting_id: int
    source: str
    created_at: datetime
    segments: list[TranscriptSegmentResponse] = []
    total_segments: int = 0

File: frontend/src/types/transcript.ts
export interface TranscriptSegment {
  id: number;
  speaker_name: string;
  start_time_ms: number;
  end_time_ms: number;
  text: string;
  sequence_index: number;
  speaker_id: number | null;
}

export interface TranscriptResponse {
  id: number;
  meeting_id: number;
  source: "seeded" | "uploaded" | "generated";
  segments: TranscriptSegment[];
  total_segments: number;
}

export interface TranscriptSearchResult {
  segment_id: number;
  speaker_name: string;
  start_time_ms: number;
  end_time_ms: number;
  highlighted_text: string;
  sequence_index: number;
}

export interface TranscriptSearchResponse {
  query: string;
  total: number;
  results: TranscriptSearchResult[];
}
